packages/cipherstash-proxy/src/proxy/mod.rs

```diff
@@ -43,96 +61,109 @@ impl Proxy {
         // Ensures error on start if credential or network issue
         zerokms.init_cipher(None).await?;
 
-        let encrypt_config = EncryptConfigManager::init(&config.database).await?;
-        // TODO: populate EqlTraitImpls based in config
-        let schema = SchemaManager::init(&config.database).await?;
-
-        let eql_version = {
-            let client = connect::database(&config.database).await?;
-            let rows = client
-                .query("SELECT eql_v2.version() AS version;", &[])
-                .await;
-
-            match rows {
-                Ok(rows) => rows.first().map(|row| row.get("version")),
-                Err(err) => {
-                    warn!(
-                        msg = "Could not query EQL version from database",
-                        error = err.to_string()
-                    );
-                    None
-                }
-            }
-        };
+        let encrypt_config_manager = EncryptConfigManager::init(&config.database).await?;
+
+        let schema_manager = SchemaManager::init(&config.database).await?;
+
+        let eql_version = Proxy::eql_version(&config).await?;
+
+        let (reload_sender, reload_receiver) = mpsc::unbounded_channel();
+
+        Proxy::receive(
+            reload_receiver,
+            schema_manager.clone(),
+            encrypt_config_manager.clone(),
+        );
 
         Ok(Proxy {
-            config,
+            config: Arc::new(config),
             zerokms,
-            encrypt_config,
-            schema,
+            encrypt_config_manager,
+            schema_manager,
             eql_version,
+            reload_sender,
         })
     }
 
+    pub async fn eql_version(config: &TandemConfig) -> Result<Option<String>, Error> {
+        let client = connect::database(&config.database).await?;
+        let rows = client
+            .query("SELECT eql_v2.version() AS version;", &[])
+            .await;
+
+        let version = match rows {
+            Ok(rows) => rows.first().map(|row| row.get("version")),
+            Err(err) => {
+                warn!(
+                    msg = "Could not query EQL version from database",
+                    error = err.to_string()
+                );
+                None
+            }
+        };
+        Ok(version)
+    }
+
+    pub fn receive(
```

Comment:
Responsible for receiving `ReloadCommand` messages. 

The `ReloadCommand` holds a `responder` that lets the calling context know that reload is complete.